app/models/spree/flow_io_product_decorator.rb

```diff
@@ -86,6 +87,20 @@ def add_flow_price_range(prices, product_zone)
       prices
     end
 
+    def adjust_zone(zone)
+      self.zone_ids ||= []
+      zone_id_string = zone.id.to_s
+      return if zone_ids.include?(zone_id_string)
+
+      self.zone_ids << zone_id_string
+      self.zone_ids = zone_ids
+      update_columns(meta: meta.to_json)
```

Comment:
Is this necessary to update the zone id?

Reply:
After rethinking and doing some tests. I will remove this call from here. 

I have another suggestion though.. This very same action is being done when importing ALL variant's data from Flow. It is not something that is currently being done automatically, though. Do you think it would be best to remove it from there as well? I honestly would prefer to avoid automatically assigning the zone there as well, but that will require us to assign a list of products to a zone manually (I think we already have a script/task to do that.. will have to check)

Reply:
Nevermind. We will need to remove this automatic assignation unfortunately from both services. Unless we have some way of avoiding synchronization for certain products (eg: hallmarked ones). Otherwise, when running the task to import all items's data we will end up assigning those hallmarked versions as well :(

Reply:
Probably not necessary for this PR, but maybe we should discuss on how to automate that assignation without being disruptive xD